Gems/Atom/Tools/MaterialCanvas/Assets/MaterialCanvas/TestData/combine_color_channels.materialcanvasnode.azasset

```diff
@@ -12,31 +12,47 @@
                 "description": "Red color channel",
                 "supportedDataTypes": [
                     "float"
-                ]
+                ],
+                "defaultValue": {
+                    "$type": "float",
+                    "Value": 1.0
+                }
             },
             {
                 "name": "inG",
                 "displayName": "G",
                 "description": "Green color channel",
                 "supportedDataTypes": [
                     "float"
-                ]
+                ],
+                "defaultValue": {
+                    "$type": "float",
+                    "Value": 1.0
+                }
             },
             {
                 "name": "inB",
                 "displayName": "B",
                 "description": "Blue color channel",
                 "supportedDataTypes": [
                     "float"
-                ]
+                ],
+                "defaultValue": {
+                    "$type": "float",
+                    "Value": 1.0
+                }
```

Comment:
In the material and material type source data formats, we found a way to load/store AZ::variant types (similar to AZ::any) without specifying a "$type" by just preserving whatever type was used in JSON to represent the value, and later converted it to the target data type. You might be able to achieve the same kind of thing here. See JsonMaterialPropertyValueSerializer, MaterialSourceData::ApplyPropertiesToAssetCreator, and MaterialAsset::Finalize. I'm not promising the same could be done here but might be worth a look.

Reply:
Ongoing discussion with the core team about getting the JSON serializer updated to natively handle AZStd any/variant. There's no easy way around this since multiple systems are going to have to serialize these values: material system, material component, graph model, graph canvas, dynamic node configuration. Other tools, like the pass editor, might require support for their own types. If the dynamic node manager (in my local changes) and the graph context are initialized with the supported datatypes before any serialization is done we might be able to use the registered data type list to help serialize or deserialize the values without the type ID. But that will not solve the problem for other systems.

The material value serializer previously required loading the material type source data or descriptor in order to know what the data type being read was.

Reply:
> The material value serializer previously required loading the material type source data or descriptor in order to know what the data type being read was.
Yes, I changed it to remove that requirement. It no longer references the material type source data. If it sees a JSON int, then it stores a int. It does any conversion on the fly later when the material type info becomes available.